include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,481 @@
+     * @param currentStep
+     */
+    void notify(uint32_t currentStep)
+    {
+        /* CORE + BORDER is PIC black magic, currently not needed
+            *
+            */
+        std::cout << "SAXS plugin enabled" << std::endl;
+        calculateSAXS < CORE + BORDER > (currentStep);
+    }
+
+    void pluginRegisterHelp(po::options_description& desc)
+    {
+            desc.add_options()
+                ((pluginPrefix + ".period").c_str(), po::value<std::string> (&notifyPeriod), "enable plugin [for each n-th step]")
+                ((pluginPrefix + ".qx_max").c_str(), po::value<float_64 > (&q_max[0])->default_value(5), "reciprocal space range qx_max (A^-1)")
+                ((pluginPrefix + ".qy_max").c_str(), po::value<float_64 > (&q_max[1])->default_value(5), "reciprocal space range qy_max (A^-1)")
+                ((pluginPrefix + ".qz_max").c_str(), po::value<float_64 > (&q_max[2])->default_value(5), "reciprocal space range qz_max (A^-1)")
+                ((pluginPrefix + ".qx_min").c_str(), po::value<float_64 > (&q_min[0])->default_value(-5), "reciprocal space range qx_min (A^-1)")
+                ((pluginPrefix + ".qy_min").c_str(), po::value<float_64 > (&q_min[1])->default_value(-5), "reciprocal space range qy_min (A^-1)")
+                ((pluginPrefix + ".qz_min").c_str(), po::value<float_64 > (&q_min[2])->default_value(-5), "reciprocal space range qz_min (A^-1)")
+                ((pluginPrefix + ".n_qx").c_str(), po::value<unsigned int> (&n_qx)->default_value(100), "Number of qx")
+                ((pluginPrefix + ".n_qy").c_str(), po::value<unsigned int> (&n_qy)->default_value(100), "Number of qy")
+                ((pluginPrefix + ".n_qz").c_str(), po::value<unsigned int> (&n_qz)->default_value(1), "Number of qz");
+
+    }
+
+
+    std::string pluginGetName() const
+    {
+        return pluginName;
+    }
+
+
+    void setMappingDescription(MappingDesc *cellDescription)
+    {
+        this->cellDescription = cellDescription;
+    }
+
+    void restart(uint32_t timeStep, const std::string restartDirectory)
+    {
+        // Keep this empty
+    }
+
+    void checkpoint(uint32_t timeStep, const std::string restartDirectory)
+    {
+        // Keep this empty
+    }
+
+
+
+
+
+private:
+    
+// static unsigned int getNq(unsigned int n_qx,unsigned int n_qy,unsigned int n_qz)
+// {
+//     unsigned int n_q = n_qx * n_qy * n_qz;
+//     return n_q;
+// }
+    // template<typename gb>
+    // void createQ(gb& q, float_X q_min, float_X q_step, unsigned int n_q)
+    // {   
+    //     for (int i = 0; i < n_q; i++)
+    //         q[i] = q_min + q_step * i;
+    // }
+
+
+    /**
+     * The plugin is loaded on every MPI rank, and therefor this function is
+     * executed on every MPI rank.
+     * One host with MPI rank 0 is defined to be the master.
+     * It creates a folder where all the
+     * results are saved and, depending on the type of radiation calculation,
```

Comment:
I think this is an old comment - perhaps it should be slightly reworded for the SAXS plugin.